cmd/slsa-github-generator/attest.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	slsav02 "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.2"
+	"github.com/spf13/cobra"
+
+	"github.com/slsa-framework/slsa-github-generator/github"
+	"github.com/slsa-framework/slsa-github-generator/signing/sigstore"
+	"github.com/slsa-framework/slsa-github-generator/slsa"
+)
+
+const provenanceOnlyBuildType = "https://github.com/slsa-framework/slsa-github-generator@v1"
+
+func parseSubjects(subjectsCSV string) ([]intoto.Subject, error) {
+	var parsed []intoto.Subject
+
+	subjects := strings.Split(subjectsCSV, ",")
+	for _, s := range subjects {
+		parts := strings.SplitN(s, "@", 2)
+		if len(parts) == 0 {
+			return nil, errors.New("missing subject name")
+		}
+		if len(parts) == 1 {
+			return nil, fmt.Errorf("expected sha256 hash for subject %q", parts[0])
+		}
+
+		parsed = append(parsed, intoto.Subject{
+			Name: parts[0],
+			Digest: slsav02.DigestSet{
+				"sha256": parts[1],
```

Comment:
suggestion: do a sanity check to verify that it looks like a sha256 hash (`^[a-fA-F0-9]{64}$`), since that could be a common mistake.

Also force it to lowercase, as required by the [spec](https://github.com/in-toto/attestation/blob/main/spec/field_types.md#DigestSet).